src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -17,10 +17,227 @@ namespace System
     /// </summary>
     public static partial class MemoryExtensions
     {
+        /// <summary>
+        /// Moves the source characters to the destination with a specified number of characters starting at a specified position removed.
+        /// </summary>
+        /// <param name="source">The source span from which the characters are removed.</param>
+        /// <param name="startIndex">The zero-based position to begin deleting characters.</param>
+        /// <param name="count">The number of characters to delete.</param>
+        /// <param name="destination">The span to copy the leftover characters into.</param>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when the destination Span is too small to contain the source Span with the characters removed
+        /// (i.e. destination.Length &lt; source.Length - count).
+        /// </exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when either <paramref name="startIndex"/> or <paramref name="count"/> is less than zero or
+        /// when <paramref name="startIndex"/> plus <paramref name="count"/> specify a position outside the source span.
+        /// </exception>
+        public static void Remove(this ReadOnlySpan<char> source, int startIndex, int count, Span<char> destination)
+        {
+            if ((uint)startIndex > (uint)source.Length || (uint)count > (uint)(source.Length - startIndex))
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+            if (destination.Length < source.Length - count)
+                ThrowHelper.ThrowArgumentException_DestinationTooShort();
+
+            // Nothing to remove
+            if (count == 0)
+            {
+                source.CopyTo(destination);
+                return;
+            }
+
+            if (source.Length - count != 0)
+            {
+                if (startIndex == 0)
+                {
+                    source.Slice(count).CopyTo(destination);
+                    return;
+                }
+
+                // destination is after source, reverse the order of copying
```

Comment:
Why do you have to do it? Isn't our CopyTo smart enough to handle overlapping ranges? If it's not, should we make it so?

Reply:
Our CopyTo is smart enough to handle overlapping ranges, but it does not help you here. This code has two CopyTo operations and they have to be ordered correctly.

Reply:
Precisely. Since we are doing multiple copies, we have to do that check ourselves. If this API was in-place remove, we could get away without having this check.

Reply:
The can be in place; it's just in common scenarios they won't be as the source will come from a string. But I can just do:
 
```c#
Span<char> s = new char[100];
Remove(s.Slice(0, 50), 40, 10, s.Slice(30));
```